evolution/engine.py

```
@@ -871,17 +912,19 @@ async def resume_from_checkpoint(self, path: str) -> None:
             raise FileNotFoundError(f"Checkpoint file not found: {path}")
 
         try:
-            with open(checkpoint_path, 'rb') as f:
+            with open(checkpoint_path, "rb") as f:
                 checkpoint_data: CheckpointData = pickle.load(f)
```

Ruff (0.14.7)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)